nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -819,7 +819,7 @@ in
       };
 
       services.zfs.zed.settings = {
-        ZED_EMAIL_PROG = lib.mkIf cfgZED.enableMail (
+        ZED_EMAIL_PROG = lib.mkIf (cfgZED.enableMail && config.services.mail.sendmailSetuidWrapper.enable) (
```

Comment:
This should be an assertion instead, otherwise we silently ignore `enableMail = true`. Especially as the default value for `enableMail` is already `config.services.mail.sendmailSetuidWrapper.enable != null`. Possibly also that default should additionally check against `.enable` (either that or the `enable` is not necessary here…either way the default and assertion should match).

I’m also curious what your config was.